tests/rptest/services/redpanda.py

```diff
@@ -832,6 +833,188 @@ def __init__(self,
+        interface matches the sample pattern against sample names, and requires
+        that exactly one (family, sample) match the query. All values for the
+        sample across the requested set of nodes are returned in a flat array.
+
+        None will be returned if less than one (family, sample) matches.
+        An exception will be raised if more than one (family, sample) matches.
+
+        Example:
+
+           The query:
+
+              redpanda.metrics_sample("under_replicated")
+
+           will return an array containing MetricSample instances for each node and
+           core/shard in the cluster. Each entry will correspond to a value from:
+
+              family = vectorized_cluster_partition_under_replicated_replicas
+              sample = vectorized_cluster_partition_under_replicated_replicas
+        """
+        pass
+
+    def signal_redpanda(self, node, signal=signal.SIGKILL, idempotent=False):
+        """
+        :param idempotent: if true, then kill-like signals are ignored if
+                           the process is already gone.
+        """
+        pass
+
+    def set_cluster_config(self,
+                           values: dict,
+                           expect_restart: bool = False,
+                           admin_client: Optional[Admin] = None,
+                           timeout: int = 10):
+        """
+        Update cluster configuration and wait for all nodes to report that they
+        have seen the new config.
+
+        :param values: dict of property name to value. if value is None, key will be removed from cluster config
+        :param expect_restart: set to true if you wish to permit a node restart for needs_restart=yes properties.
+                               If you set such a property without this flag, an assertion error will be raised.
+        """
+        pass
+
+    def node_id(self, node, force_refresh=False, timeout_sec=30):
+        """
+        Returns the node ID of a given node. Uses a cached value unless
+        'force_refresh' is set to True.
+
+        NOTE: this is not thread-safe.
+        """
+        pass
+
+    def rolling_restart_nodes(self,
+                              nodes,
+                              override_cfg_params=None,
+                              start_timeout=None,
+                              stop_timeout=None,
+                              use_maintenance_mode=True,
+                              omit_seeds_on_idx_one=True):
+        pass
+
+    def metric_sum(self,
+                   metric_name,
+                   metrics_endpoint: MetricsEndpoint = MetricsEndpoint.METRICS,
+                   ns=None,
+                   topic=None):
+        """
+        Pings the 'metrics_endpoint' of each node and returns the summed values
+        of the given metric, optionally filtering by namespace and topic.
+        """
+        pass
+
+    def partitions(self, topic):
```

Comment:
Wouldn't this method be the same for any backend?  It's just using Kafka API

Reply:
i've moved this impl into the base class with 3466c44baa2cb59e8172f8c6b900209b7a7cece0

Reply:
update: i've force-pushed and this is now 8cbffed5285c76bf888d5c15fcd9eae451231119